src/Magnum/GL/Buffer.h

```diff
@@ -766,7 +766,7 @@ class MAGNUM_GL_EXPORT Buffer: public AbstractObject {
          *      WebGL 1.0, see particular @ref Target values for version
          *      requirements.
          */
-        static void bind(Target target, UnsignedInt firstIndex, std::initializer_list<std::tuple<Buffer*, GLintptr, GLsizeiptr>> buffers);
+        static void bind(Target target, UnsignedInt firstIndex, Containers::ArrayView<std::tuple<Buffer*, GLintptr, GLsizeiptr>> buffers);
```

Comment:
Ah, yes, it would be nice to get rid of the 30k+LOC from `std::tuple` as well... Nowadays there's [Containers::Triple](https://doc.magnum.graphics/corrade/classCorrade_1_1Containers_1_1Triple.html) that could be used instead.

The original `std::initializer_list<std::tuple>` signature needs to be still present though, for backwards compatibility, and delegate to the new variant. Unfortunately `std::tuple` has the internal order of elements reversed so just an `arrayCast()` won't cut it, the backwards compat overload would need to allocate a new array.

The old signature would look like this, inside it you'll have to allocate a new Array, copy the elements over and call with it into the new bind() overload.

```cpp
        #ifdef MAGNUM_BUILD_DEPRECATED
        /** 
         * @m_deprecated_since_latest Use @ref bind(Target, UnsignedInt, std::initializer_list<Containers::Triple<Buffer*, GLintptr, GLsizeiptr>>) 
         *      instead.
         */
        #endif
        static CORRADE_DEPRECATED("use the Containers::Triple overload instead") void bind(Target target, UnsignedInt firstIndex, std::initializer_list<std::tuple<Buffer*, GLintptr, GLsizeiptr>> buffers);
```

And then the tuple forward declaration can be moved down to the conditional as well:

```cpp
#ifdef MAGNUM_BUILD_DEPRECATED
...
/* For deprecated bind(..., std::initializer_list<std::tuple>) */
#include <Corrade/Utility/StlForwardTuple.h>
#endif
```

And same for `#include <tuple>` if it appears inside `Buffer.cpp` or elsewhere.

Oh, and as I said for the other overload, a `std::initializer_list` variant needs to be still present as well. Similarly, just delegating:

```cpp
 void bind(Target target, UnsignedInt firstIndex, std::initializer_list<Containers::Triple<Buffer*, GLintptr, GLsizeiptr>> buffers) {
     return bind(target, firstIndex, Containers::arrayView(buffers);
 }
```

You may need to deinline it into the cpp file to not have to include the Triple header. Also there *may* be an issue with the `std::tuple` and `Containers::Triple` variants being ambiguous if one passes stuff like `{{&buffer, offset, size}}` to them, but hopefully not. At least not on C++11, where one had to use `std::make_tuple()` instead of `{}`. If you hit such corner case either in the tests here or in the Wonderland codebase, please let me know, I'll try to figure out a way to resolve the ambiguity.

Reply:
I'm doing only the `std::initializer_list` change in a first commit, and will do the `std::tuple` change afterwards